src/taipy/core/common/_mongo_connector.py

```diff
@@ -0,0 +1,54 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+from typing import Dict, Optional
+
+import pymongo
+
+
+def _connect_mongodb(
```

Comment:
not sure if this makes sense to put under a lru cache function as it will require extra step from user, tried to wrap it within another function as well but tests are failing. I'll have a look at it but not sure if it will be on time for the release. The effect is (probably) minimal as of now if not many users are using mongoDN so might not be a big issue after all

Reply:
Well I tried but it failed for unknown reason :/ and so I added lru on enterprise while keeping the code the same on core. Still investigating the issue